connector/connect/src/test/scala/org/apache/spark/sql/connect/planner/SparkConnectProtoSuite.scala

```diff
@@ -215,4 +180,16 @@ class SparkConnectProtoSuite extends PlanTest with SparkConnectPlanTest {
     }
     proto.Relation.newBuilder().setLocalRelation(localRelationBuilder.build()).build()
   }
+
+  private def analyzePlan(plan: LogicalPlan): LogicalPlan = {
+    val connectAnalyzed = analysis.SimpleAnalyzer.execute(plan)
+    analysis.SimpleAnalyzer.checkAnalysis(connectAnalyzed)
+    EliminateSubqueryAliases(connectAnalyzed)
```

Comment:
Why do we need to do this? Would be great to add a comment here to explain.

Reply:
+1

Reply:
I added some comments to clarify this function's usage.

Reply:
but why do we need to eliminate subquery alias?

Reply:
hmmm this is what I borrowed from https://github.com/apache/spark/blob/c50d865fa9eb5207bc8c9992e37843412ec0cbc3/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/dsl/package.scala#L513

We are using this Catalyst DSL analyze call already before this refactoring.

Reply:
Did we hit any issues if not doing it?

Reply:
There is no issue after removing it. I pushed a commit to remove it anyway.